From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 17 Dec 2015 17:37:09 +0000
Subject: [armhf,sparc64] Force ZONE_DMA to be enabled
Forwarded: not-needed

CONFIG_ZONE_DMA is supposed to be selected according to the architecture
configuration, not by drivers.

In 4.3.3 several sound drivers were changed to depend on, rather than
selecting, ZONE_DMA.  This disabled it and thus broke ABI for most
arm{el,hf}, mips* and sparc64 flavours.

Version 4.3.1-1 FTBFS on armel and didn't finish building on mips* in
time to include an ABI reference for 4.3.3-1.  So the mips* flavours
have quietly changed ABI (oops) while armhf and sparc now FTBFS.
Leave mips* as it is, but force ZONE_DMA on to reverse the ABI change
for armhf and sparc64.

---
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -204,6 +204,7 @@ config ARCH_MAY_HAVE_PC_FDC
 
 config ZONE_DMA
 	bool
+	default y if ARCH_MULTI_V7
 
 config NEED_DMA_MAP_STATE
        def_bool y
--- a/arch/sparc/Kconfig
+++ b/arch/sparc/Kconfig
@@ -131,7 +131,7 @@ config HIGHMEM
 
 config ZONE_DMA
 	bool
-	default y if SPARC32
+	default y
 
 config NEED_DMA_MAP_STATE
 	def_bool y
